pkgs/applications/misc/squeekboard/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, cmake
+, meson
+, ninja
+, pkgconfig
+, gnome3
+, glib
+, gtk3
+, wayland
+, wayland-protocols
+, rustc
+, cargo
+, libcroco
+, libxml2
+, libxkbcommon
+, rustPlatform
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "squeekboard";
+  version = "1.9.1";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "334898c5a5d45a5883654b4f2f07f28ac8f2bfc4";
+    sha256 = "1nrihpffr91p3ipif5k5z9vcvkwxk6qy8hp93w13s7gnwl2g93hx";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
+    rustc
+    cargo
+    glib.dev
+    gtk3.dev
+    wayland
+    gnome3.gnome-desktop
```

Comment:
Could you add a comment for which build tools are these required? Since they are primarily libraries. (It will be helpful when we get to splitting the packages.)

Reply:
I don't what you mean. This is the output I get without gnome-desktop:

```
Message:
------------------
squeekboard 1.8.0

------------------

Program cargo found: YES (/nix/store/qqx1fbkv0i1a2qn6fll27gas6p8hs32b-cargo-1.42.0/bin/cargo)
Program cargo.sh found: YES (/build/source/cargo.sh)
Program cargo_build.sh found: YES (/build/source/cargo_build.sh)
Found pkg-config: /nix/store/nc3ivxhlqyj2alck8p47srvwzzgkbh2i-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Using 'PKG_CONFIG_PATH' from environment with value: '/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/afwkyx0h1whgggbz4gclx6d6iyz4sv2w-zlib-1.2.11-dev/lib/pkgconfig:/nix/store/27s8lpl9c50scx0936c31j44yi7cda1k-libffi-3.3-dev/lib/pkgconfig:/nix/store/q6ar7s3ld5l537pkjymwyz92a1cgv1qg-gtk+3-3.24.20-dev/lib/pkgconfig:/nix/store/bjg5yljgxf88l9lfnfags2p9w670rwxy-at-spi2-atk-2.34.2/lib/pkgconfig:/nix/store/18qmdsyy9f6xxxcxbk8zi7k7gl04z3r7-atk-2.36.0-dev/lib/pkgconfig:/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/n2amx937mh6k39iadvh567kq4g2ljly7-cairo-1.16.0-dev/lib/pkgconfig:/nix/store/qx8p9zq8wdhaysnhgb7912y8pjdj1b3c-fontconfig-2.12.6-dev/lib/pkgconfig:/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/lib/pkgconfig:/nix/store/5afwqbvxly5scbid8j1yybanr18k3cl4-bzip2-1.0.6.0.1-dev/lib/pkgconfig:/nix/store/fj4qycc9fnn9pkkmk4vmg68q1961icb7-libpng-apng-1.6.37-dev/lib/pkgconfig:/nix/store/jqki871ikmaq4s6glrbn28cvfzp8ghkj-expat-2.2.8-dev/lib/pkgconfig:/nix/store/5n9qf33f4ynngqjl4fi61ks9ccqhgbgi-pixman-0.38.4/lib/pkgconfig:/nix/store/pg9xi8kcc00h6h9i585sxcbqhcsl69h3-libXext-1.3.4-dev/lib/pkgconfig:/nix/store/kz68jkipag56bsn1jv559v1yd2kql45x-xorgproto-2019.1/share/pkgconfig:/nix/store/gd5mi1mvxklc1gsjyb1i75dw5x5hrclp-libXau-1.0.9-dev/lib/pkgconfig:/nix/store/crkk522qcbji0nyhf89fabzvbj5n5dcg-libXrender-0.9.10-dev/lib/pkgconfig:/nix/store/6w463c8dh0qzf5l2lz42k683nl46bkp9-libX11-1.6.8-dev/lib/pkgconfig:/nix/store/mdp6m40d5yxjvjlgchlxzmr7g31c0mnp-libxcb-1.13.1-dev/lib/pkgconfig:/nix/store/f7z5p96yyq1rbpgkys3pn4lr2cz3d84m-xcb-util-0.4.0-dev/lib/pkgconfig:/nix/store/khr2giczjm79mzgy23dg7brdd8ii650x-libGL-1.3.1-dev/lib/pkgconfig:/nix/store/2w4kby6pnscgdsbay9xzb2nrifcjml1r-libglvnd-1.3.1-dev/lib/pkgconfig:/nix/store/xqvccn03aln6irpr3pi1akma4nacm5qf-fribidi-1.0.7/lib/pkgconfig:/nix/store/pj515z2qsfhwsp3izx0paxfj434hwjqn-gdk-pixbuf-2.40.0-dev/lib/pkgconfig:/nix/store/svf57d8nwwhsc5dpkl30ndvb6b420x1y-libtiff-4.1.0-dev/lib/pkgconfig:/nix/store/78zkj5rkgm140n3rvmxall7qn613jhh5-libjpeg-turbo-2.0.4-dev/lib/pkgconfig:/nix/store/qf5d4n9s8ra2jkbni5ad0dyv984rkxwy-xz-5.2.5-dev/lib/pkgconfig:/nix/store/vhc7j53yqsvrr35whr178bh7spxpan6n-gsettings-desktop-schemas-3.36.1/share/pkgconfig:/nix/store/5cbkfrh2x52mfd9jjhl1acm4qjcyds1i-libICE-1.0.10-dev/lib/pkgconfig:/nix/store/zc2cvq0kzfqf03s2dcqv2bpmfkwbzfhm-libSM-1.2.3-dev/lib/pkgconfig:/nix/store/7nd75qgnvg6glxmih6y17d9qnsqy63nv-libXcomposite-0.4.5-dev/lib/pkgconfig:/nix/store/77vs7c9xxwvssxkjg0xp3622jddkbagm-libXfixes-5.0.3-dev/lib/pkgconfig:/nix/store/wmplwklka4wz2zp4xf5nmn7zm4n59i1v-libXcursor-1.2.0-dev/lib/pkgconfig:/nix/store/pbvirs5nyai88crwycmzgz1srz7jpssm-libXi-1.7.10-dev/lib/pkgconfig:/nix/store/31l8dqrnbv156nly52yc6p8iifr1cwj3-libXrandr-1.5.2-dev/lib/pkgconfig:/nix/store/hnigkkd0xdyl0byb09xxmx55r502acgs-pango-1.44.7-dev/lib/pkgconfig:/nix/store/kicxnsx8wx5vdjdm1hxq13kkpykkb067-harfbuzz-2.6.4-dev/lib/pkgconfig:/nix/store/arwz5zkwqln2z2fzd56sv04kvz4jzqwc-graphite2-1.3.14/lib/pkgconfig:/nix/store/s9zs326ikm627czkwcc0si5q9nsyf5s0-libXft-2.3.3-dev/lib/pkgconfig:/nix/store/4hlj2zavxvm4khyyj5m2fsbh7gck6y0r-mesa-20.0.2-dev/lib/pkgconfig:/nix/store/m2nzplawf621cj2rr13sjgxhglikw486-libXdamage-1.1.5-dev/lib/pkgconfig:/nix/store/67p8vwjj53w02kjwxkks5nqrkrkzjyyi-libXxf86vm-1.1.4-dev/lib/pkgconfig:/nix/store/akcqxqkxivvdx7p17napb0hv7i3hpkv4-libdrm-2.4.100-dev/lib/pkgconfig:/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/lib/pkgconfig:/nix/store/wck1wg7m434kq8j33lcfsly90rdilrz7-wayland-protocols-1.20/share/pkgconfig:/nix/store/8297h0ilzpgnvk448aiydgpvzi3jijv4-libXinerama-1.1.4-dev/lib/pkgconfig:/nix/store/32wa7ann5bq7vl61dffcqx2v0kgp01px-gmp-6.2.0-dev/lib/pkgconfig:/nix/store/yr1pc96a75l0923hv6lfm70l7bz5ca6x-libxml2-2.9.10-dev/lib/pkgconfig:/nix/store/bwg82bn5m8kgncq390ic4yzg7qia7rih-libxkbcommon-0.10.0-dev/lib/pkgconfig'
Found pkg-config: /nix/store/nc3ivxhlqyj2alck8p47srvwzzgkbh2i-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Using 'PKG_CONFIG_PATH' from environment with value: '/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/afwkyx0h1whgggbz4gclx6d6iyz4sv2w-zlib-1.2.11-dev/lib/pkgconfig:/nix/store/27s8lpl9c50scx0936c31j44yi7cda1k-libffi-3.3-dev/lib/pkgconfig:/nix/store/q6ar7s3ld5l537pkjymwyz92a1cgv1qg-gtk+3-3.24.20-dev/lib/pkgconfig:/nix/store/bjg5yljgxf88l9lfnfags2p9w670rwxy-at-spi2-atk-2.34.2/lib/pkgconfig:/nix/store/18qmdsyy9f6xxxcxbk8zi7k7gl04z3r7-atk-2.36.0-dev/lib/pkgconfig:/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/n2amx937mh6k39iadvh567kq4g2ljly7-cairo-1.16.0-dev/lib/pkgconfig:/nix/store/qx8p9zq8wdhaysnhgb7912y8pjdj1b3c-fontconfig-2.12.6-dev/lib/pkgconfig:/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/lib/pkgconfig:/nix/store/5afwqbvxly5scbid8j1yybanr18k3cl4-bzip2-1.0.6.0.1-dev/lib/pkgconfig:/nix/store/fj4qycc9fnn9pkkmk4vmg68q1961icb7-libpng-apng-1.6.37-dev/lib/pkgconfig:/nix/store/jqki871ikmaq4s6glrbn28cvfzp8ghkj-expat-2.2.8-dev/lib/pkgconfig:/nix/store/5n9qf33f4ynngqjl4fi61ks9ccqhgbgi-pixman-0.38.4/lib/pkgconfig:/nix/store/pg9xi8kcc00h6h9i585sxcbqhcsl69h3-libXext-1.3.4-dev/lib/pkgconfig:/nix/store/kz68jkipag56bsn1jv559v1yd2kql45x-xorgproto-2019.1/share/pkgconfig:/nix/store/gd5mi1mvxklc1gsjyb1i75dw5x5hrclp-libXau-1.0.9-dev/lib/pkgconfig:/nix/store/crkk522qcbji0nyhf89fabzvbj5n5dcg-libXrender-0.9.10-dev/lib/pkgconfig:/nix/store/6w463c8dh0qzf5l2lz42k683nl46bkp9-libX11-1.6.8-dev/lib/pkgconfig:/nix/store/mdp6m40d5yxjvjlgchlxzmr7g31c0mnp-libxcb-1.13.1-dev/lib/pkgconfig:/nix/store/f7z5p96yyq1rbpgkys3pn4lr2cz3d84m-xcb-util-0.4.0-dev/lib/pkgconfig:/nix/store/khr2giczjm79mzgy23dg7brdd8ii650x-libGL-1.3.1-dev/lib/pkgconfig:/nix/store/2w4kby6pnscgdsbay9xzb2nrifcjml1r-libglvnd-1.3.1-dev/lib/pkgconfig:/nix/store/xqvccn03aln6irpr3pi1akma4nacm5qf-fribidi-1.0.7/lib/pkgconfig:/nix/store/pj515z2qsfhwsp3izx0paxfj434hwjqn-gdk-pixbuf-2.40.0-dev/lib/pkgconfig:/nix/store/svf57d8nwwhsc5dpkl30ndvb6b420x1y-libtiff-4.1.0-dev/lib/pkgconfig:/nix/store/78zkj5rkgm140n3rvmxall7qn613jhh5-libjpeg-turbo-2.0.4-dev/lib/pkgconfig:/nix/store/qf5d4n9s8ra2jkbni5ad0dyv984rkxwy-xz-5.2.5-dev/lib/pkgconfig:/nix/store/vhc7j53yqsvrr35whr178bh7spxpan6n-gsettings-desktop-schemas-3.36.1/share/pkgconfig:/nix/store/5cbkfrh2x52mfd9jjhl1acm4qjcyds1i-libICE-1.0.10-dev/lib/pkgconfig:/nix/store/zc2cvq0kzfqf03s2dcqv2bpmfkwbzfhm-libSM-1.2.3-dev/lib/pkgconfig:/nix/store/7nd75qgnvg6glxmih6y17d9qnsqy63nv-libXcomposite-0.4.5-dev/lib/pkgconfig:/nix/store/77vs7c9xxwvssxkjg0xp3622jddkbagm-libXfixes-5.0.3-dev/lib/pkgconfig:/nix/store/wmplwklka4wz2zp4xf5nmn7zm4n59i1v-libXcursor-1.2.0-dev/lib/pkgconfig:/nix/store/pbvirs5nyai88crwycmzgz1srz7jpssm-libXi-1.7.10-dev/lib/pkgconfig:/nix/store/31l8dqrnbv156nly52yc6p8iifr1cwj3-libXrandr-1.5.2-dev/lib/pkgconfig:/nix/store/hnigkkd0xdyl0byb09xxmx55r502acgs-pango-1.44.7-dev/lib/pkgconfig:/nix/store/kicxnsx8wx5vdjdm1hxq13kkpykkb067-harfbuzz-2.6.4-dev/lib/pkgconfig:/nix/store/arwz5zkwqln2z2fzd56sv04kvz4jzqwc-graphite2-1.3.14/lib/pkgconfig:/nix/store/s9zs326ikm627czkwcc0si5q9nsyf5s0-libXft-2.3.3-dev/lib/pkgconfig:/nix/store/4hlj2zavxvm4khyyj5m2fsbh7gck6y0r-mesa-20.0.2-dev/lib/pkgconfig:/nix/store/m2nzplawf621cj2rr13sjgxhglikw486-libXdamage-1.1.5-dev/lib/pkgconfig:/nix/store/67p8vwjj53w02kjwxkks5nqrkrkzjyyi-libXxf86vm-1.1.4-dev/lib/pkgconfig:/nix/store/akcqxqkxivvdx7p17napb0hv7i3hpkv4-libdrm-2.4.100-dev/lib/pkgconfig:/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/lib/pkgconfig:/nix/store/wck1wg7m434kq8j33lcfsly90rdilrz7-wayland-protocols-1.20/share/pkgconfig:/nix/store/8297h0ilzpgnvk448aiydgpvzi3jijv4-libXinerama-1.1.4-dev/lib/pkgconfig:/nix/store/32wa7ann5bq7vl61dffcqx2v0kgp01px-gmp-6.2.0-dev/lib/pkgconfig:/nix/store/yr1pc96a75l0923hv6lfm70l7bz5ca6x-libxml2-2.9.10-dev/lib/pkgconfig:/nix/store/bwg82bn5m8kgncq390ic4yzg7qia7rih-libxkbcommon-0.10.0-dev/lib/pkgconfig'
Run-time dependency wayland-protocols found: YES 1.20
Program wayland-scanner found: YES (/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/bin/wayland-scanner)
Configuring config.h using configuration
src/meson.build:7: WARNING: Got an empty configuration_data() object and found no substitutions in the input file 'config.h.in'. If you want to copy a file to the build dir, use the 'copy:' keyword argument added in 0.47.0
Using 'PKG_CONFIG_PATH' from environment with value: '/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/afwkyx0h1whgggbz4gclx6d6iyz4sv2w-zlib-1.2.11-dev/lib/pkgconfig:/nix/store/27s8lpl9c50scx0936c31j44yi7cda1k-libffi-3.3-dev/lib/pkgconfig:/nix/store/q6ar7s3ld5l537pkjymwyz92a1cgv1qg-gtk+3-3.24.20-dev/lib/pkgconfig:/nix/store/bjg5yljgxf88l9lfnfags2p9w670rwxy-at-spi2-atk-2.34.2/lib/pkgconfig:/nix/store/18qmdsyy9f6xxxcxbk8zi7k7gl04z3r7-atk-2.36.0-dev/lib/pkgconfig:/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/lib/pkgconfig:/nix/store/n2amx937mh6k39iadvh567kq4g2ljly7-cairo-1.16.0-dev/lib/pkgconfig:/nix/store/qx8p9zq8wdhaysnhgb7912y8pjdj1b3c-fontconfig-2.12.6-dev/lib/pkgconfig:/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/lib/pkgconfig:/nix/store/5afwqbvxly5scbid8j1yybanr18k3cl4-bzip2-1.0.6.0.1-dev/lib/pkgconfig:/nix/store/fj4qycc9fnn9pkkmk4vmg68q1961icb7-libpng-apng-1.6.37-dev/lib/pkgconfig:/nix/store/jqki871ikmaq4s6glrbn28cvfzp8ghkj-expat-2.2.8-dev/lib/pkgconfig:/nix/store/5n9qf33f4ynngqjl4fi61ks9ccqhgbgi-pixman-0.38.4/lib/pkgconfig:/nix/store/pg9xi8kcc00h6h9i585sxcbqhcsl69h3-libXext-1.3.4-dev/lib/pkgconfig:/nix/store/kz68jkipag56bsn1jv559v1yd2kql45x-xorgproto-2019.1/share/pkgconfig:/nix/store/gd5mi1mvxklc1gsjyb1i75dw5x5hrclp-libXau-1.0.9-dev/lib/pkgconfig:/nix/store/crkk522qcbji0nyhf89fabzvbj5n5dcg-libXrender-0.9.10-dev/lib/pkgconfig:/nix/store/6w463c8dh0qzf5l2lz42k683nl46bkp9-libX11-1.6.8-dev/lib/pkgconfig:/nix/store/mdp6m40d5yxjvjlgchlxzmr7g31c0mnp-libxcb-1.13.1-dev/lib/pkgconfig:/nix/store/f7z5p96yyq1rbpgkys3pn4lr2cz3d84m-xcb-util-0.4.0-dev/lib/pkgconfig:/nix/store/khr2giczjm79mzgy23dg7brdd8ii650x-libGL-1.3.1-dev/lib/pkgconfig:/nix/store/2w4kby6pnscgdsbay9xzb2nrifcjml1r-libglvnd-1.3.1-dev/lib/pkgconfig:/nix/store/xqvccn03aln6irpr3pi1akma4nacm5qf-fribidi-1.0.7/lib/pkgconfig:/nix/store/pj515z2qsfhwsp3izx0paxfj434hwjqn-gdk-pixbuf-2.40.0-dev/lib/pkgconfig:/nix/store/svf57d8nwwhsc5dpkl30ndvb6b420x1y-libtiff-4.1.0-dev/lib/pkgconfig:/nix/store/78zkj5rkgm140n3rvmxall7qn613jhh5-libjpeg-turbo-2.0.4-dev/lib/pkgconfig:/nix/store/qf5d4n9s8ra2jkbni5ad0dyv984rkxwy-xz-5.2.5-dev/lib/pkgconfig:/nix/store/vhc7j53yqsvrr35whr178bh7spxpan6n-gsettings-desktop-schemas-3.36.1/share/pkgconfig:/nix/store/5cbkfrh2x52mfd9jjhl1acm4qjcyds1i-libICE-1.0.10-dev/lib/pkgconfig:/nix/store/zc2cvq0kzfqf03s2dcqv2bpmfkwbzfhm-libSM-1.2.3-dev/lib/pkgconfig:/nix/store/7nd75qgnvg6glxmih6y17d9qnsqy63nv-libXcomposite-0.4.5-dev/lib/pkgconfig:/nix/store/77vs7c9xxwvssxkjg0xp3622jddkbagm-libXfixes-5.0.3-dev/lib/pkgconfig:/nix/store/wmplwklka4wz2zp4xf5nmn7zm4n59i1v-libXcursor-1.2.0-dev/lib/pkgconfig:/nix/store/pbvirs5nyai88crwycmzgz1srz7jpssm-libXi-1.7.10-dev/lib/pkgconfig:/nix/store/31l8dqrnbv156nly52yc6p8iifr1cwj3-libXrandr-1.5.2-dev/lib/pkgconfig:/nix/store/hnigkkd0xdyl0byb09xxmx55r502acgs-pango-1.44.7-dev/lib/pkgconfig:/nix/store/kicxnsx8wx5vdjdm1hxq13kkpykkb067-harfbuzz-2.6.4-dev/lib/pkgconfig:/nix/store/arwz5zkwqln2z2fzd56sv04kvz4jzqwc-graphite2-1.3.14/lib/pkgconfig:/nix/store/s9zs326ikm627czkwcc0si5q9nsyf5s0-libXft-2.3.3-dev/lib/pkgconfig:/nix/store/4hlj2zavxvm4khyyj5m2fsbh7gck6y0r-mesa-20.0.2-dev/lib/pkgconfig:/nix/store/m2nzplawf621cj2rr13sjgxhglikw486-libXdamage-1.1.5-dev/lib/pkgconfig:/nix/store/67p8vwjj53w02kjwxkks5nqrkrkzjyyi-libXxf86vm-1.1.4-dev/lib/pkgconfig:/nix/store/akcqxqkxivvdx7p17napb0hv7i3hpkv4-libdrm-2.4.100-dev/lib/pkgconfig:/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/lib/pkgconfig:/nix/store/wck1wg7m434kq8j33lcfsly90rdilrz7-wayland-protocols-1.20/share/pkgconfig:/nix/store/8297h0ilzpgnvk448aiydgpvzi3jijv4-libXinerama-1.1.4-dev/lib/pkgconfig:/nix/store/32wa7ann5bq7vl61dffcqx2v0kgp01px-gmp-6.2.0-dev/lib/pkgconfig:/nix/store/yr1pc96a75l0923hv6lfm70l7bz5ca6x-libxml2-2.9.10-dev/lib/pkgconfig:/nix/store/bwg82bn5m8kgncq390ic4yzg7qia7rih-libxkbcommon-0.10.0-dev/lib/pkgconfig'
Run-time dependency gio-2.0 found: YES 2.64.1
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency gnome-desktop-3.0 found: NO (tried pkgconfig)

src/meson.build:36:0: ERROR: Dependency "gnome-desktop-3.0" not found, tried pkgconfig
```

Reply:
Then gnome-desktop should likely go to `buildInputs`, not `nativeBuildInputs`. You can try to use `strictDeps = true;` to enforce the distinction better.